components/scream/src/dynamics/homme/atmosphere_dynamics_rayleigh_friction.cpp

```diff
@@ -14,8 +14,12 @@ void HommeDynamics::rayleigh_friction_init()
   using KT = KokkosTypes<DefaultDevice>;
   using Pack = ekat::Pack<Real,N>;
 
+  const int nlevs = m_dyn_grid->get_num_vertical_levels();
+  const auto npacks= ekat::PackInfo<Pack::n>::num_packs(nlevs);
+  m_otau = decltype(m_otau)("otau", npacks);
+
   // Rayleigh friction paramaters
-  m_rayk0     = m_params.get<int>("Rayleigh Friction Vertical Level", 2);
+  m_rayk0     = m_params.get<int>("Rayleigh Friction Vertical Level", 1);
```

Comment:
@ambrad I went ahead and converted this to c++ indexing (set this `[0,numlevs-1)`). Let me know if you would like it reverted back. I verified that the profile is the same as before (which matches what you posted in scream slack from F90)

Reply:
That's a matter of taste. @AaronDonahue @PeterCaldwell do we use 0-based or 1-based indexing in user-facing parameters?

Separately, we're no longer using PL entries of the form "Rayleigh Friction Vertical Level"; switch to something like Rayleigh_friction_vertical_level, where I purpose capitalize Rayleigh since it's a name and lowercase everything else.

Reply:
Since we went with snake_case, I would keep lowercase also for names, acronyms, and whatnots. But again, taste.

Reply:
uggg, thats a tricky one.  I'm inclined to say we want to stick with 1-based since most users will be familiar with that.  Although the use of the variable inside C++ should be 0-based to avoid confusion when developing or just plain reading the code.  If we stick with outward 1-based can we just subtract 1 from what we read from the parameter list?  also would be good to put a comment into the code explaining what is going on here.  Also a comment in the default parameter list...

Reply:
@bartgol I'd like to disagree re: case. Snake case (vs. camel case) is the best convention b/c you can capitalize what should be cap'ed and nothing else.

Reply:
Fair point.

Reply:
That said, I'm realizing this is potentially going to cause user experience issues, so all lowercase is fine. We can use a capital letter if a situation arises where actual ambiguity can occur due to constant-naming conventions.

Reply:
RE: 0- vs 1- based, I changed the param to be 1-based, then immediately convert to 0-based with a comment. Let me know what y'all think.